Wikipedia/UI-V5/WMFRelatedTitleListDataSource.m

```diff
@@ -0,0 +1,165 @@
+    numberOfExtractCharacters:(NSUInteger)numberOfExtractCharacters
+                      fetcher:(WMFRelatedSearchFetcher*)fetcher {
+    self = [super initWithItems:nil];
+    if (self) {
+        self.title                     = title;
+        self.dataStore                 = dataStore;
+        self.savedPageList             = savedPageList;
+        self.numberOfExtractCharacters = numberOfExtractCharacters;
+        self.relatedSearchFetcher      = fetcher;
+
+        self.cellClass = [WMFArticlePreviewCell class];
+
+        @weakify(self);
+        self.cellConfigureBlock = ^(WMFArticlePreviewCell* cell,
+                                    MWKArticle* article,
+                                    UICollectionView* collectionView,
+                                    NSIndexPath* indexPath) {
+            @strongify(self);
+            MWKRelatedSearchResult* searchResult = self.relatedTitleResults[indexPath.item];
+            cell.title           = article.title;
+            cell.descriptionText = searchResult.wikidataDescription;
+            cell.image           = [article bestThumbnailImage];
+            [cell setSummaryHTML:searchResult.extractHTML fromSite:self.title.site];
+        };
+        [self fetch];
+    }
+    return self;
+}
+
+- (void)setCollectionView:(UICollectionView*)collectionView {
+    [super setCollectionView:collectionView];
+    [self.collectionView registerNib:[WMFArticlePreviewCell wmf_classNib]
+          forCellWithReuseIdentifier:[WMFArticlePreviewCell identifier]];
+}
+
+#pragma mark - Fetching
+
+- (void)fetch {
+    @weakify(self);
+    [self.relatedSearchFetcher fetchArticlesRelatedToTitle:self.title
+                                  numberOfExtactCharacters:self.numberOfExtractCharacters
+                                               resultLimit:WMFMaxRelatedSearchResultLimit]
+    .then(^(WMFRelatedSearchResults* searchResults) {
+        @strongify(self);
+        NSMutableArray* mutableResults = [searchResults.results mutableCopy];
+        NSArray* items = [mutableResults bk_reduce:[NSMutableArray arrayWithCapacity:self.relatedTitleResults.count]
+                                         withBlock:^(NSMutableArray* articles,
+                                                     MWKRelatedSearchResult* relatedSearchResult) {
+            MWKTitle* title = [[MWKTitle alloc] initWithString:relatedSearchResult.displayTitle
+                                                          site:self.title.site];
+            NSError* error;
+            NSDictionary* resultJSON = [MTLJSONAdapter JSONDictionaryFromModel:relatedSearchResult error:&error];
+            if (!resultJSON) {
+                DDLogError(@"Unexpected error re-serializing search result %@. Error: %@",
+                           relatedSearchResult, error);
+                [mutableResults removeObject:relatedSearchResult];
+                return articles;
+            }
+            MWKArticle* article = [[MWKArticle alloc]
```

Comment:
I think we want to move away from the article creation here as we talked about yesterday. Seems like this may be a good opportunity to refactor the articleDataSource to be a titleDataSource instead to prevent us from doing this here.